test/deadline_adaptor_for_unreal/unit/UnrealClient/step_handlers/test_render_step_handler.py

```diff
@@ -0,0 +1,71 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+import sys
+import pytest
+from unittest.mock import MagicMock
+
+unreal_mock = MagicMock()
+unreal_mock.log.return_value = MagicMock()
+
+sys.modules['unreal'] = unreal_mock
```

Comment:
I wonder if it might make more sense to wrap our logging calls in some other method which could either test for importing unreal and fall back on prints or something if it's not around, or we could patch that wrapper instead, or maybe there's some other better solution I'm not thinking of.  Seems like needing to do this all over in our tests isn't the best solution long term.  What do you think?   If we did decide this was the best approach currently we should comment what's going on at least I think

Reply:
In most of the submitter we use a lot of unreal objects, methods and classes. The unreal Python module does not contain any logic, only declarations. Therefore, it is not advisable to mock specific objects and methods used when converting MRQ jobs to OpenJob. In this case, we test methods that do not rely on unreal objects (we managed to get by with two mocked classes MRQJob and ShotInfo), and logging is the only thing used there. Therefore, this is more of a single case than a reason to come up with a solution for logging in all tests (unreal.log is the most trivial thing that can be there)

Reply:
Ok - we've got a couple cases of it happening in this CR so figured it would be worth considering now, but yes if you think it's unlikely to need to be applied in many other cases maybe that's not such an issue.  For the logging case at least it seems like we should be able to install a customer logger when the unreal module is available and then just route all of our logging through python logging - but yeah if we do think this is a one time issue maybe that's outside the scope of what we're really doing here.